python-checks/src/test/resources/checks/genericFunctionTypeParameter.py

```diff
@@ -0,0 +1,51 @@
+
+from typing import TypeVar
+
+def non_compliant_1():
+    _T = TypeVar("_T", bound=str)
+    #    ^^^^^^^^^^^^^^^^^^^^^^^^>  {{"TypeVar" is assigned here.}}
+    def func(a: _T, b: int) -> str:  # Noncompliant {{Use a generic type parameter for this function instead of a "TypeVar".}}
+    #   ^^^^    ^^<                                 {{Use of "TypeVar" here.}}
+        ...
+def non_compliant_2():
+    _T = TypeVar("_T", bound=str)
+    #    ^^^^^^^^^^^^^^^^^^^^^^^^>     {{"TypeVar" is assigned here.}}
+
+    def func(a: _T, b: int) -> _T:  # Noncompliant {{Use a generic type parameter for this function instead of a "TypeVar".}}
+    #   ^^^^    ^^<                                {{Use of "TypeVar" here.}}
+    #                          ^^@-1<              {{Use of "TypeVar" here.}}
+        ...
+
+def non_compliant_3():
+    _T = TypeVar("_T", bound=str)
+    #    ^^^^^^^^^^^^^^^^^^^^^^^^>  {{"TypeVar" is assigned here.}}
+    _R = TypeVar("_R")
+    #    ^^^^^^^^^^^^^>             {{"TypeVar" is assigned here.}}
+    _S = TypeVar("_S")
+    #    ^^^^^^^^^^^^^>             {{"TypeVar" is assigned here.}}
+    def func(a: _T, b: _R) -> _S:  # Noncompliant {{Use a generic type parameter for this function instead of a "TypeVar".}}
+    #   ^^^^    ^^<                               {{Use of "TypeVar" here.}}
+    #                  ^^@-1<                     {{Use of "TypeVar" here.}}
+    #                         ^^@-2<              {{Use of "TypeVar" here.}}
+        ...
+
+def non_compliant_4():
+    from typing import TypeVar as TV
+    _T = TV("_T", bound=str)
+    #    ^^^^^^^^^^^^^^^^^^^>  {{"TypeVar" is assigned here.}}
+    def func(a: _T, b: int) -> str:  # Noncompliant {{Use a generic type parameter for this function instead of a "TypeVar".}}
+    #   ^^^^    ^^<                                 {{Use of "TypeVar" here.}}
+        ...
+
+def FN_non_compliant():
+    _T = TypeVar("_T", bound=str)
+
+    def func(a: list[_T]) -> set[_T]:  # FN
```

Comment:
Please don't forget to reference a ticket here (you can reuse the same example in the ticket) so that the FN is fully documented and we can eventually do something about it.

Reply:
Done in f6edd0239d1600ea93f64f6d104abbc4c17bdb59.